### PR TITLE
Fix Issue 271: [counting of epoch & trained size in resume training does not mat…

### PR DESCRIPTION
**Pull request by psyyz10**
## What changes were proposed in this pull request?

Fix [counting of epoch & trained size in resume training does not match original counts]

## How was this patch tested?

Train a model and save checkpoint by each epoch. Reload a snapshot, see the epoch and iteration are consistent with the saved checkpoint.

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/271

### CODE REVIEW

**Comment by yiheng** [comment]
can you add an unit test for this fix?

**Comment by psyyz10** [comment]
Done with unit tests

**Comment by yiheng** [comment]
LGTM

File: dl/src/test/scala/com/intel/analytics/bigdl/optim/DistriOptimizerSpec.scala [hunk 1]
```diff
@@ -243,6 +246,30 @@ class DistriOptimizerSpec extends FlatSpec with Matchers with BeforeAndAfter {
     batchNormalization.runningVar.storage().array() should be (
       Array(1188.2811870277535, 1188.2811870277535)
     )
+  }
+
+  "DistriOptimizer checkpoint" should "work correclty" in {
+    val path = "/tmp/model/"
```
Comment (yiheng) [inline dl/src/test/scala/com/intel/analytics/bigdl/optim/DistriOptimizerSpec.scala]: please do not hard code the path

Comment (psyyz10) [inline dl/src/test/scala/com/intel/analytics/bigdl/optim/DistriOptimizerSpec.scala]: Done

**Pull request merged**